itest/lnd_onchain_test.go

```diff
@@ -731,3 +731,106 @@ func genAnchorSweep(ht *lntest.HarnessTest,
 
 	return btcutil.NewTx(tx)
 }
+
+// testRemoveTx tests that we are able to remove an unconfirmed transaction
+// from the internal wallet as long as the tx is still unconfirmed. This test
+// also verifies that after the tx is removed (while unconfirmed) it will show
+// up as confirmed as soon as the original transaction is mined.
+func testRemoveTx(ht *lntest.HarnessTest) {
+	// Create a new node so that we start with no funds on the internal
+	// wallet.
+	alice := ht.NewNode("Alice", nil)
+
+	const initialWalletAmt = btcutil.SatoshiPerBitcoin
+
+	// Funding the node with an initial balance.
+	ht.FundCoins(initialWalletAmt, alice)
+
+	// Create an address for Alice to send the coins to.
+	req := &lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_WITNESS_PUBKEY_HASH,
+	}
+	resp := alice.RPC.NewAddress(req)
+
+	// We send half the amount to that address generating two unconfirmed
+	// outpoints in our internal wallet.
+	sendReq := &lnrpc.SendCoinsRequest{
+		Addr:   resp.Address,
+		Amount: initialWalletAmt / 2,
+	}
+	alice.RPC.SendCoins(sendReq)
+	txID := ht.Miner.AssertNumTxsInMempool(1)[0]
+
+	// Make sure the unspent number of utxos is 2 and the unconfirmed
+	// balances add up.
+	unconfirmed := ht.GetUTXOsUnconfirmed(
+		alice, lnwallet.DefaultAccountName,
+	)
+	require.Lenf(ht, unconfirmed, 2,
+		"expected 2 unconfirmed tx, got %v", len(unconfirmed))
+
+	// Get the raw transaction to remove it later.
+	tx := ht.Miner.GetNumTxsFromMempool(1)[0]
+
+	// Calculate the tx fee so we can compare the end amounts. We are
+	// sending from the internal wallet to the internal wallet so only
+	// the tx fee applies when calucalting the final amount of the wallet.
+	txFee := ht.CalculateTxFee(tx)
+
+	// All of alice's balance is unconfirmed and equals the initial amount
+	// minus the tx fee.
+	aliceBalResp := alice.RPC.WalletBalance()
+	require.Equal(ht, btcutil.Amount(initialWalletAmt)-txFee,
```

Comment:
nit: can use `require.EqualValues` to remove some of the type casts, then maybe things fit on a single line (if not, IMO we should format this like normal multi-line function calls, not the `fmt.Errorf()` or `log.XXXf()` exceptions).

Reply:
what about `require.NoErrorf` ? I guess it makes sense to keep the formating there  equal to fmt.Errorf() ?

Reply:
Yeah, the formatting guide isn't super clear about that and we probably use both cases in the existing itest code. I would argue for: if it's a call with a format string (e.g. a call to `require.X...f()`) the same rule as for `fmt.Errorf()` applies. If not, we try to use the normal multi-line function call formatting.
But maybe that's too specific, feel free to ignore.